std/algorithm/sorting.d

```diff
@@ -2626,6 +2626,120 @@ unittest
     }
 }
 
+// Always leaves median in r[2]. Credit goes to Xinok, see e.g.
+// http://stackoverflow.com/questions/11065963/possible-to-partition-five-elements-by-median-with-six-comparisons
+private void medianOf5(alias less, Range)(Range r)
+{
+    assert(r.length == 5);
```

Comment:
This use case is expressly what contracts are for.

Plus, it saves a tiny bit of run time cost when using `-release`, which is helpful in functions like this where it will be called many times in a performance critical piece of code.